llvm/docs/DTLTO.rst

```diff
@@ -60,165 +80,126 @@ Distributors are programs responsible for:
-                "-o", ["primary_output", 0],
-                "-c", "-x", "ir", ["primary_input", 0],
-                ["summary_index", "-fthinlto-index=", 0],
-                "--target=x86_64-sie-ps5"
-            ]
+            "args": ["clang.exe", "-c", "-flto", "-fprofile-sample-use=my.prof"],
+            "inputs": ["my.prof"]
         },
         "jobs": [
             {
-                "primary_input": ["dtlto.o"],
-                "summary_index": ["dtlto.1.51232.native.o.thinlto.bc"],
-                "primary_output": ["dtlto.1.51232.native.o"],
-                "imports": [],
-                "additional_inputs": ["my.profdata"]
+                "args": ["t1.bc", "-fthinlto-index=t1.1.7944.native.o.thinlto.bc", "-o", "t1.1.7944.native.o", "-fproc-stat-report=t1.1.7944.stats.txt"],
+                "inputs": ["t1.bc", "t1.1.7944.native.o.thinlto.bc"],
+                "outputs": ["t1.1.7944.native.o", "t1.1.7944.stats.txt"]
+            },
+            {
+                "args": ["t2.bc", "-fthinlto-index=t2.2.7944.native.o.thinlto.bc", "-o", "t2.2.7944.native.o", "-fproc-stat-report=t2.2.7944.stats.txt"],
+                "inputs": ["t2.bc", "t2.2.7944.native.o.thinlto.bc"],
+                "outputs": ["t2.2.7944.native.o", "t2.2.7944.stats.txt"]
             }
         ]
     }
 
-Key Features of the Schema
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+Each entry in the ``jobs`` array represents a single backend compilation job.
+Each job object records its own command-line arguments and input/output files.
+Shared inputs and arguments are defined once in the ``common`` object.
 
-- **Input/Output Paths**: Paths are stored in per-file-type array fields. This
-  allows files to be adjusted, if required, to meet the constraints of the
-  underlying distribution system. For example, a system may only be able to read
-  and write remote files to `C:\\sandbox`. The remote paths used can be adjusted
-  by the distributor for such constraints. Once outputs are back on the local
-  system, the distributor can rename them as required.
+Reserved Entries:
 
+- The first entry in the ``common.args`` array specifies the compiler
+  executable to invoke.
+- The first entry in each job's ``inputs`` array is the bitcode file for the
+  module being compiled.
+- The second entry in each job's ``inputs`` array is the corresponding
+  individual summary index file.
+- The first entry in each job's ``outputs`` array is the primary output object
+  file.
 
-- **Command-Line Template**: Command-line options are stored in a common
-  template to avoid duplication for each job. The template consists of an array
-  of strings and arrays. The arrays are placeholders which reference per-job
-  paths. This allows the remote compiler and its arguments to be changed without
-  updating the distributors.
+An array-of-strings representation is used for command-line arguments to avoid
+issues with quoting and shell escaping.
```

Comment:
Is this statement necessary? And there might still be "issues" depending on what the "executor" itself needs to do...

Reply:
I'll remove it. The less to review the better.